bootstrapvz/plugins/puppet/tasks.py

```diff
@@ -36,28 +52,75 @@ def run(cls, info):
             raise TaskError(msg)
 
 
-class AddPackages(Task):
-    description = 'Add puppet package'
+class AddPuppetlabsPC1SourcesList(Task):
+    description = 'Adding Puppetlabs APT repo to the list of sources.'
     phase = phases.preparation
 
     @classmethod
     def run(cls, info):
-        info.packages.add('puppet')
+        if (info.manifest.release == jessie):
+            info.source_lists.add('puppetlabs', 'deb http://apt.puppetlabs.com jessie PC1')
+        if (info.manifest.release == wheezy):
+            info.source_lists.add('puppetlabs', 'deb http://apt.puppetlabs.com wheezy PC1')
+
+
+class InstallPuppetlabsPC1ReleaseKey(Task):
+    description = 'Install puppetlabs PC1 Release key into the keyring'
+    phase = phases.package_installation
+
+    @classmethod
+    def run(cls, info):
+        key_file_name = 'DEB-GPG-KEY-puppet'
```

Comment:
@andsens  At the time I didnt manage to get it working that way, so i tried the other approach by fetching the key from the internet itself. the reason why the package ca-certificates was added is to ensure the wget command would succeed and still verify the SSL connection to the apt.puppetlabs.com repo.